### DISCUSSION TITLE
Serialize and Deserialize MetaData Object

### DISCUSSION THREAD

**Discussion by ajaykushwaha** [sqlalchemy/sqlalchemy#10552] [Usage Questions]
I have a requirement to dynamically create the Metadata of an existing database without making direct requests to the database. 

There's an API that should provide the Metadata Object of a Database. However, the challenge is that the requests are stateless, making it impossible to retain the object across different calls. Performing 'metadata.reflect' every time a user requests the metadata object is not feasible due to concerns about the availability of the database and the potential burden on its resources.

To address this issue, I am considering saving the Metadata Object. One approach is to extract all tables, their columns, and relevant attributes such as data types, nullability, primary keys, etc., and store this information in JSON or another suitable format. This stored Metadata Object can then be used to dynamically create the metadata without requiring direct access to the database.

However, I am interested in exploring alternative methods for achieving this. I anticipate that there might be some information loss during the extraction and storage process. Could you please provide insights into the potential extent of this information loss

**Comment by CaselIT** [comment]
Hi,

I'm not sure I 100% understand your use case. If the db is always the same for all users, I don't understand why you have to dynamically reflect it.

As for serialization there are tests for metadata and table pickle support, so that form of serialization should be supported.
If you want to serializing things in json, considering that you are reflecting the tables, you may look into the internals of the reflection logic, where all the data that's used to reflect a db is stored inside a `_ReflectionInfo` instance, that should not be too hard to serialize to json. The issue is that it's a private api at the moment, and re-creating the tables from scratch takes a bit of time even if you already have all the data. Overall if you can use it, pickle is probably a better solution
